src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -945,7 +945,11 @@ private static void CheckBaseAddress(Uri? baseAddress, string parameterName)
 
             if (!HttpUtilities.IsHttpUri(baseAddress))
             {
+#if TARGETS_BROWSER
```

Comment:
I'd like to avoid the proliferation of such ifdefs.  We've generally tried to do this with partial files, e.g. this would call a method `ThrowInvalidUril` which would be defined appropriately in the relevant build.

Alternatively, I'd be fine if we just changed the shared error message to be `The 'blah' scheme is not supported.` rather than having two different messages that list what is supported.

Reply:
I like the more detailed one so left it in but moved the message definition to partial file as described.